### PR TITLE
Add API for ProxySQL UI-Server

### PR DESCRIPTION
**Pull request by TasdidurRahman**
Signed-off-by: TasdidurRahman <tasdid@appscode.com>

### CODE REVIEW

**Review by spectro30 — DISMISSED** [review]
LGTM

File: apis/ui/v1alpha1/proxysql_insight_types.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+	AbortedClientConnections   *int32          `json:"abortedClientConnections,omitempty"`
+	ConnectedClientConnections *int32          `json:"connectedClientConnections,omitempty"`
+	CreatedClientConnections   *int32          `json:"createdClientConnections,omitempty"`
+	AbortedServerConnections   *int32          `json:"abortedServerConnections,omitempty"`
+	ConnectedServerConnections *int32          `json:"connectedServerConnections,omitempty"`
+	CreatedServerConnections   *int32          `json:"createdServerConnections,omitempty"`
+	QueryTypeInsight           []RuleExecution `json:"queryInsight,omitempty"`
+}
+
+//type PodInsights struct {
```
Comment (pkbhowmick) [inline apis/ui/v1alpha1/proxysql_insight_types.go:46]: Please consider removing commented code if not necessary.

File: apis/ui/v1alpha1/proxysql_insight_types.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ResourceKindProxySQLInsight = "ProxySQLInsight"
+	ResourceProxySQLInsight     = "proxysqlinsight"
+	ResourceProxySQLInsights    = "proxysqlinsights"
+)
+
+type RuleExecution struct {
```
Comment (pkbhowmick) [inline apis/ui/v1alpha1/proxysql_insight_types.go:28]: Short description/comment for this field would be really helpful.

File: apis/ui/v1alpha1/proxysql_insight_types.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+type ProxySQLInsightSpec struct {
+	Version                string       `json:"version"`
+	Status                 string       `json:"status"`
+	MaxConnections         *int32       `json:"maxConnections,omitempty"`
+	LongQueryTimeThreshold *float64     `json:"longQueryTimeThreshold,omitempty"`
+	PodInsights            []PodInsight `json:"podInsights,omitempty"`
+}
+
+// ProxySQLInsight is the Schema for the proxysqlinsights API
+
```
Comment (pkbhowmick) [inline apis/ui/v1alpha1/proxysql_insight_types.go:61]: If we want to keep `ProxySQLInsight is the Schema for the proxysqlinsights API` as a documentation for ProxySQLInsight struct then please remove this extra line. Comment out this line will also work I think.

**Review by pkbhowmick — CHANGES_REQUESTED** [review]

File: apis/ui/v1alpha1/proxysql_insight_types.go (hunk 1 above)
Comment (TasdidurRahman) [reply apis/ui/v1alpha1/proxysql_insight_types.go:46]: removed

File: apis/ui/v1alpha1/proxysql_insight_types.go (hunk 3 above)
Comment (TasdidurRahman) [reply apis/ui/v1alpha1/proxysql_insight_types.go:61]: fixed

File: apis/ui/v1alpha1/proxysql_insight_types.go (hunk 2 above)
Comment (TasdidurRahman) [reply apis/ui/v1alpha1/proxysql_insight_types.go:30]: added

**Review by tamalsaha — APPROVED** [review]

**Pull request merged**